store/tikv/latch/latch.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package latch
+
+import (
+	"fmt"
+	"math/bits"
+	"sort"
+	"sync"
+
+	"github.com/spaolacci/murmur3"
+)
+
+// Latch stores a key's waiting transactions information.
+type Latch struct {
+	hasWaiting bool
+	// The startTS of the transaction which is the head of waiting transactions.
+	head        uint64
+	maxCommitTS uint64
+	sync.Mutex
+}
+
+// acquire tries to get current key's lock for the transaction with startTS.
+// success is true when success
+// timeout is true when the startTS is already timeout
+// newWait is true when current transaction is new for the current latch
+func (l *Latch) acquire(startTS uint64) (success, timeout, newWait bool) {
+	l.Lock()
+	defer l.Unlock()
+
+	if timeout = startTS <= l.maxCommitTS; timeout {
+		return
+	}
+
+	if l.hasWaiting == false {
+		l.head = startTS
+		l.hasWaiting = true
+		newWait = true
+	}
+	success = l.head == startTS
```

Comment:
Seems `newWait` and `success` will always be the same, as long as we don't `acquire` multiple times for a same ts.

Reply:
I also confused by `newWait`